typegen/examples/polkadot.rs

```diff
@@ -0,0 +1,53 @@
+use std::marker::PhantomData;
+
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use scale_info::PortableRegistry;
+use scale_typegen::{TypeGenerator, TypeGeneratorSettings};
+use serde_json::Value;
+use syn::parse_quote;
+
+#[allow(unused)]
+pub struct DecodedBits<Store, Order> {
+    bits: scale_bits::Bits,
+    _marker: PhantomData<(Store, Order)>,
+}
+
+/// This example shows how to use metadata from a polkadot node to generate rust types.
+///
+/// Some types might need
+pub fn main() {
+    let type_registry = read_registry_from_json();
+    let settings = TypeGeneratorSettings::default()
+        .type_mod_name("my_types")
+        .decoded_bits_type_path(parse_quote!(DecodedBits));
+
+    let type_generator = TypeGenerator::new(&type_registry, &settings);
+    let code = type_generator
+        .generate_types_mod()
+        .unwrap()
+        .to_token_stream();
+
+    write_pretty_tokens(code, "./src/polkadot.rs");
+}
+
+fn read_registry_from_json() -> PortableRegistry {
+    let json: Value =
+        serde_json::from_str(&std::fs::read_to_string("./artifacts/polkadot.json").unwrap())
+            .unwrap();
+    let types = json.as_array().unwrap()[1]
+        .as_object()
+        .unwrap()
+        .get("V15")
+        .unwrap()
+        .get("types")
+        .unwrap();
+    let registry: PortableRegistry = serde_json::from_value(types.clone()).unwrap();
+    registry
+}
```

Comment:
Personally I'd load some SCALE encoded `polkadot.scale` metadata and then pull the type registry from that (the SCALe encoded version feels like the more "common" one to be using in Rust land)

Reply:
Same, then we could just decode the scale file and say `md.types`